engine/examples/dotnetcore.md

```diff
@@ -0,0 +1,102 @@
+  constructing
+  your solutions
+
+## Prerequisites
+This example assumes you already have an ASP.NET Core app 
+on your machine. If you are new to ASP.NET you can follow a [simple 
+tutorial](https://www.asp.net/get-started) to initialize a project or clone our [ASP.NET Docker Sample](https://github.com/dotnet/dotnet-docker-samples/tree/master/aspnetapp).
+
+## Create a Dockerfile for an ASP.NET Core application
+1.  Create a `Dockerfile` in your project folder.
+2.  Add the text below to your `Dockerfile` for either Linux or [Windows
+   Containers](https://docs.microsoft.com/en-us/virtualization/windowscontainers/about/).
+    The tags below are multi-arch meaning they will pull either Windows or 
+    Linux containers depending on what mode is set in [Docker for 
+Windows](https://docs.docker.com/docker-for-windows/). Read more on [switching containers](https://docs.docker.com/docker-for-windows/#switch-between-windows-and-linux-containers).
+3.  The `Dockerfile` assumes that your application is called `aspnetapp`. Change
+   the `Dockerfile` to use the DLL file of your project.
+
+```dockerfile
+FROM microsoft/aspnetcore-build:1.1 AS build-env
+WORKDIR /app
+
+# Copy csproj and restore as distinct layers
+COPY *.csproj ./
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
+RUN dotnet publish -c Release -o out
+
+# Build runtime image
+FROM microsoft/aspnetcore:1.1
+WORKDIR /app
+COPY --from=build-env /app/out .
+ENTRYPOINT ["dotnet", "aspnetapp.dll"]
+```
+
+4.  To make your build context as small as possible add a [`.dockerignore`
+   file](https://docs.docker.com/engine/reference/builder/#dockerignore-file) 
+   to
+   your project folder and copy the following into it.
+
+```dockerignore
+bin\
+obj\
+```
+
+## Build and run the Docker image
+1.  In the command prompt and navigate to your project folder.
+2.  Use the following commands to build and run your Docker image:
+
+```console
+docker build -t aspnetapp .
+docker run -d -p 80:80 aspnetapp
+```
+
+## View the web page running from a container
+* Go to [localhost:80](http://localhost:80) to access your app in a web browser.
+* If you are using the Nano [Windows
+  Container](https://docs.docker.com/docker-for-windows/) and have not updated
+   to the Windows Creator Update there is a bug affecting how [Windows 10 
+   talks to Containers via
+  "NAT"](https://github.com/Microsoft/Virtualization-Documentation/issues/181#issuecomment-252671828)
+  (Network Address Translation). You must hit the IP of the container
+  directly. You can get the IP address of your container with the following
+  steps:
+  1.  Run `docker ps` and copy the hash of your container ID.
+  2.  Run `docker inspect -f "{{ .NetworkSettings.Networks.nat.IPAddress }}"
+     HASH` where `HASH` is replaced with your container ID.
+  3.  Copy the container ip address and paste into your browser with port 80.
+     (Ex: 172.16.240.197:80)
```

Comment:
We generally use `e.g.` instead of `Ex:`; "(e.g., 172.16.240.197:80)"

Also may be worth emphasising the part to copy, either making it bold, or inside back tics